### PR TITLE
Fix PlasmaMax sound component crashes

### PR DESCRIPTION
**Pull request by dpogue**
I'm happy to drop the third and fourth commits if you prefer them not to be included. The fourth one in particular probably needs more investigation to figure out what to do (otoh, it happens on exit, so leaking memory maybe isn't a concern?)

### CODE REVIEW

**Comment by Hoikas** [comment]
The second commit looks fine to me. I'll look at the first and third in more detail when I'm next at a PC.

File: Sources/Tools/MaxComponent/plMaxWaveUtils.cpp [hunk 1]
```diff
@@ -62,3 +68,23 @@ SegmentMap *GetWaveSegmentMap(const MCHAR* file, plErrorMsg *pErrMsg)
...
+        waveFile.Open(M2ST(file), nullptr, WAVEFILE_READ);
+
+        return waveFile.GetHeader().fNumChannels;
+    } else {
+        plSoundBuffer* buf = new plSoundBuffer(fileName);
+        if (buf != nullptr && buf->IsValid())
+            return buf->GetHeader().fNumChannels;
+        return 0;
+    }
+}
```
Comment (Hoikas) [inline Sources/Tools/MaxComponent/plMaxWaveUtils.cpp:90]: Newline at EOF.

File: Sources/Tools/MaxComponent/plMaxWaveUtils.cpp [hunk 2]
```diff
@@ -62,3 +68,23 @@ SegmentMap *GetWaveSegmentMap(const MCHAR* file, plErrorMsg *pErrMsg)
...
+uint16_t GetWaveNumChannels(const MCHAR* file)
+{
+    const plFileName fileName = M2ST(file);
+    plFileInfo fileInfo(fileName);
+    if (!fileInfo.Exists())
+        return 0;
+
+    if (fileName.GetFileExt().compare_ni("wav", 3) == 0) {
+        CWaveFile waveFile;
+        waveFile.Open(M2ST(file), nullptr, WAVEFILE_READ);
```
Comment (Hoikas) [inline Sources/Tools/MaxComponent/plMaxWaveUtils.cpp:81]: Why not:

```suggestion
        waveFile.Open(fileName, nullptr, WAVEFILE_READ);
```

File: Sources/Tools/MaxExport/plExportDlg.cpp [hunk 3]
```diff
@@ -89,7 +89,7 @@ class plExportDlgImp : public plExportDlg
 
 public:
     plExportDlgImp();
-    ~plExportDlgImp();
+    ~plExportDlgImp() { }
```
Comment (Hoikas) [inline Sources/Tools/MaxExport/plExportDlg.cpp:92]:
```suggestion
```

File: Sources/Tools/MaxComponent/plAutoUIBase.cpp [hunk 4]
```diff
@@ -59,6 +59,7 @@ plAutoUIBase::~plAutoUIBase()
 {
     if (fDesc)
     {
+#if 0
```
Comment (Hoikas) [inline Sources/Tools/MaxComponent/plAutoUIBase.cpp:62]: We should probably strip this commit off and figure out the real problem.

**Comment by dpogue** [comment]
updated

**Comment by Hoikas** [comment]
What do you see? An unmerged-but-approved PR. Fixing...

**Pull request merged**